### PR TITLE
Remove unnecessary deepcopy in redis storage

### PR DESCRIPTION
**Pull request by wattlebirdaz**
## Motivation
Remove unnecessary deepcopy in redis storage for better performance.

## Description of the changes
Removed unnecessary deepcopy in redis storage.

### CODE REVIEW

**Comment by c-bata** [comment]
Thanks for the pull request. Changes looks good to me. I'd say we can simply remove `test_get_all_trials_deepcopy_option` to fix CI failure.

```
__________________ test_get_all_trials_deepcopy_option[redis] __________________
storage_mode = 'redis'
    @pytest.mark.parametrize("storage_mode", STORAGE_MODES)
    def test_get_all_trials_deepcopy_option(storage_mode: str) -> None:
        with StorageSupplier(storage_mode) as storage:
            study_summaries, study_to_trials = _setup_studies(storage, n_study=2, n_trial=5, seed=49)
            for study_id in study_summaries:
                with patch("copy.deepcopy", wraps=copy.deepcopy) as mock_object:
                    trials0 = storage.get_all_trials(study_id, deepcopy=True)
>                   assert mock_object.call_count > 0
E                   AssertionError: assert 0 > 0
E                    +  where 0 = <MagicMock name='deepcopy' id='[139](https://github.com/optuna/optuna/runs/6892145689?check_suite_focus=true#step:8:140)722682714600'>.call_count
```

**Comment by c-bata** [comment]
@not522 Could you review this PR?

**Review by c-bata — APPROVED** [review]
LGTM!

File: optuna/storages/_redis.py [hunk 1]
```diff
@@ -691,12 +691,7 @@ def get_all_trials(
         states: Optional[Container[TrialState]] = None,
     ) -> List[FrozenTrial]:
 
-        trials = self._get_trials(study_id, states, set())
-
-        if deepcopy:
-            return copy.deepcopy(trials)
-        else:
-            return trials
+        return self._get_trials(study_id, states, set())
```
Comment (not522) [inline optuna/storages/_redis.py:694]: Could you add a comment on why `deepcopy` is not necessary?

**Review by not522 — COMMENTED** [review]
Basically LGTM. Could you check my comment?

File: optuna/storages/_redis.py (hunk 1 above)
Comment (wattlebirdaz) [reply optuna/storages/_redis.py:695]: Added. [581f448](https://github.com/optuna/optuna/pull/3672/commits/581f448ab53587c573b105288c1994b072f6d683)

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**